rocketchat-common/src/main/java/com/rocketchat/common/listener/StreamCollectionListener.java

```diff
@@ -0,0 +1,11 @@
+package com.rocketchat.common.listener;
+
+import org.json.JSONObject;
+
+public interface StreamCollectionListener<T> {
+    void onAdded(String documentKey, T document);
+
+    void onChanged(String documentKey, JSONObject values);
```

Comment:
this should be a `void onChanged(String documentKey, T values);`

Reply:
Sometimes values contains only one key, it is not convenient to create Document object which contains just one non null data member. Getting values in the form of jsonObject is more simple and easier to understand. You do not have to check for null values in JSONObject.

Reply:
Why `onAdded` has T and `onChanged` has `JSONObject`? I believe we should use the same type everywhere.

Reply:
Hi @rafaelks, it is as I explained earlier. Let me give you example. Lets say you have UserDocument with 9-10 attributes, one of them is status. When status gets changed, server sends reply with id and status:offline. That's it. No other data, it is not feasible to construct object representing just a state and not having any other data inside it. I think representing it in the form of JSON is more appropriate for better usability. ( PS. It might look little different, but for the sake of better handling values, we need it).

Reply:
On Android we can just update the present fields on the object onto the database, I believe that on most Java frameworks would be the same...
Or we expose everything as models, or everything as JSONObjects... Some of both is confuse...